src/select-query-parser/result.ts

```diff
@@ -297,7 +297,9 @@ type ProcessSpreadNode<
 /**
  * Helper type to process the result of a spread node.
  */
-type ProcessSpreadNodeResult<Result> = ExtractFirstProperty<Result> extends infer SpreadedObject
+type ProcessSpreadNodeResult<Result> = Result extends Record<string, SelectQueryError<string>>
+  ? Result
+  : ExtractFirstProperty<Result> extends infer SpreadedObject
```

Comment:
**remark**

Trying to reproduce on our database, I encountered a case with the spread operator where an error within the spread wasn't properly handled here was the previous result for such case:

<img width="570" alt="Screenshot 2024-10-30 at 00 14 32" src="https://github.com/user-attachments/assets/9b7019cb-fc80-401a-ae68-bf7c1d31af22">